code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -85,11 +85,41 @@
 		. = if_no_id	//to prevent null-names making the mob unclickable
 	return
 
-//gets ID card object from special clothes slot or null.
-/mob/living/carbon/human/get_idcard()
+//Gets ID card from a human. If hand_first is false the one in the id slot is prioritized, otherwise inventory slots go first.
+/mob/living/carbon/human/get_idcard(hand_first = TRUE)
+	//Check hands
+	var/obj/item/card/id/id_card
+	var/obj/item/held_item
+	held_item = get_active_held_item()
+	if(held_item) //Check active hand
+		id_card = held_item.GetID()
+	if(!id_card) //If there is no id, check the other hand
+		held_item = get_inactive_held_item()
+		id_card = held_item.GetID()
+		
+	if(id_card)
+		if(hand_first)
+			return id_card
+		else
+			. = id_card
+			
+	//Check inventory slots		
 	if(wear_id)
-		return wear_id.GetID()
-
+		id_card = wear_id.GetID()
+		if(id_card)
+			return wear_id.GetID()
+	else if(belt)
+		id_card = belt.GetID()
+		if(id_card)
+			return belt.GetID()
+	else if(l_store)
+		id_card = l_store.GetID()
+		if(id_card)
+			return l_store.GetID()
+	else if(r_store)
+		id_card = r_store.GetID()
+		if(id_card)
+			return r_store.GetID()

```

Comment:
You're still calling `GetID()` twice every time

Reply:
Something went wrong with the commit then, i'll redo it